include/fizzy/fizzy.h

```diff
@@ -11,8 +11,87 @@
 extern "C" {
 #endif
 
+/// The opaque data type representing a module.
+struct fizzy_module;
+
+/// The opaque data type representing an instance (instantiated module).
+struct fizzy_instance;
+
+/// The data type representing numeric values.
+///
+/// i64 member is used to represent values of both i32 and i64 type.
+union fizzy_value
+{
+    uint64_t i64;
+    float f32;
+    double f64;
+};
+
+/// Result of execution of a function.
+typedef struct fizzy_execution_result
+{
+    /// Whether execution ended with a trap.
+    bool trapped;
+    /// Whether function returned a value. Valid only if trapped == false
+    bool has_value;
+    /// Value returned from a function. Valid only if has_value == true
+    union fizzy_value value;
+} fizzy_execution_result;
+
+
+/// Pointer to external function.
+///
+/// @param context      Opaque pointer to execution context.
+/// @param instance     Pointer to module instance.
+/// @param args         Pointer to the argument array.
+/// @param args_size   Size of the  argument array.
+/// @param depth        Call stack depth.
+typedef struct fizzy_execution_result (*fizzy_external_fn)(void* context,
+    struct fizzy_instance* instance, const union fizzy_value* args, uint32_t args_size, int depth);
```

Comment:
The `uint32_t` will be annoying to use, let's use `size_t` for array sizes for now (as in `fizzy_validate`, `fizzy_parse`, etc).

```suggestion
    struct fizzy_instance* instance, const union fizzy_value* args, size_t args_size, int depth);
```